hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/ozoneimpl/BackgroundContainerDataScanner.java

```diff
@@ -87,27 +87,30 @@ public void scanContainer(Container<?> c)
     ContainerData containerData = c.getContainerData();
     long containerId = containerData.getContainerID();
     logScanStart(containerData);
-    ScanResult result = c.scanData(throttler, canceler);
-
-    // Metrics for skipped containers should not be updated.
-    if (result.getFailureType() == DELETED_CONTAINER) {
-      LOG.error("Container [{}] has been deleted.",
-          containerId, result.getException());
-      return;
-    }
-    if (!result.isHealthy()) {
-      LOG.error("Corruption detected in container [{}]. Marking it UNHEALTHY.",
-          containerId, result.getException());
-      boolean containerMarkedUnhealthy = controller.markContainerUnhealthy(containerId, result);
-      if (containerMarkedUnhealthy) {
-        metrics.incNumUnHealthyContainers();
+    DataScanResult result = c.scanData(throttler, canceler);
```

Comment:
Nit: Not part of this PR but the whole `scanContainer` should have a metric around time taken

Reply:
NIT:
```
    if (volume.isFailed()) {
      shutdown("The volume has failed.");
      return;
    }
``` 
We should log volume info in the log message in line 80.

Reply:
The full message with the volume info is constructed and logged at INFO level by the `shutdown` method. Is there something extra you think we should add to that message?

Reply:
For the metrics, there are numerous improvements. I filed HDDS-11593 for this.